dynamic.js

```diff
@@ -59,7 +60,7 @@ module.exports = function (fastify, opts, next) {
     // Base swagger info
     // this info is displayed in the swagger file
     // in the same order as here
-    swaggerObject.swagger = '2.0'
+    servers ? swaggerObject.openapi = '3.0.0' : swaggerObject.swagger = '2.0'
```

Comment:
OAS 3.0 is not supported by this plugin and so the `servers` syntax.

So this change would not be enough to support it

Reply:
true. It worked for my use-case but as mcollina pointed out, this has to be thoroughly accounted for all the features that may be in use in the swagger 2.0 [currently looking into it]

Reply:
I was analysing how to solve the `$ref`, and I found this module [@apidevtools/swagger-parser](https://www.npmjs.com/package/@apidevtools/swagger-parser).

I think it could be useful to check the difference between the OAS2 and OAS3 formats.

Then, I thought what could be the macro-steps for this:

1. collect all the schemas
1. build one single schema (I working on it to support $ref and shared schemas)
1. translate a JSON schema _(draft 07 - since draft2019 is not broadly supported yet)_ to Swagger/OAS2.0 or OAS3 based on user options
1. validate the generated OAS object
1. render it